### PR TITLE
Security: validate and sanitize PasskeyEnrollSubmission.name (finding 3.2)  #23

### LINKED ISSUE #23: Security: validate and sanitize PasskeyEnrollSubmission.name (finding 3.2)
## Background

Security review finding **3.2** — the user-supplied `name` field in `PasskeyEnrollSubmission` is stored in `passkeys` without any length or character validation.

## Risk

- **Oversized input:** an arbitrarily long name can waste storage or trigger column-length errors at the database level (ugly 500 instead of a clean 400).
- **Malformed data in the UI:** if the name is rendered in the central-ui without escaping, a crafted name can be a stored XSS vector.

## Proposed fix

In `ConversationService.finishPasskeyEnroll`, before persisting the passkey, validate `submission.name`:

```scala
val name = submission.name.trim
if name.isEmpty || name.length > 64 || !name.matches("[\\p{L}\\p{N} _\\-'.]+")
then ZIO.fail(ConversationError.IllegalState)
else // proceed
```

Adjust the allowed character set and max length to match product requirements.

## Scope / tasks

- [ ] Define the allowed character set and maximum length (suggest: ≤ 64 chars, printable Unicode letters/digits/common punctuation)
- [ ] Add the validation guard in `finishPasskeyEnroll`
- [ ] Return `ConversationResult.IllegalState` (or a dedicated validation error) on failure
- [ ] Add tests: empty name, name > max length, disallowed characters, valid name

### PR DESCRIPTION
**Pull request by BKJN1**
Closes #23

Validate and sanitize PasskeyEnrollSubmission.name (finding 3.2)

What changed
In ConversationService.finishPasskeyEnroll, added validation of the user-supplied name field before persisting the passkey. If name is Some(value), it is trimmed and checked against:

non-empty after trim
max length of 64 characters
allowed characters: Unicode letters, digits, spaces, and _, -, ', .

None is accepted as-is (the field is optional).
Returns ConversationResult.IllegalState on validation failure, before any external service calls.

Tests added

empty name → IllegalState
name longer than 64 chars → IllegalState
name with disallowed characters (e.g. <script>) → IllegalState
valid name ("My MacBook") → completes successfully

### CODE REVIEW

**Comment by BKJN1** [comment]
@Augment review

**Comment by BKJN1** [comment]
@Augment review

File: auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala [hunk 1]
```diff
@@ -308,6 +308,68 @@ object PasskeyConversationServiceSpec extends UnitSpecBase:
...
+          _ <- env.configService.getPasskeySettings.succeedsWith(Some(passkeySettings))
+          _ <- env.webAuthnService.finishRegistration.succeedsWith(dummyPasskey)
+          _ <- env.authPropertyGenerator.nextAuthorizationCode.succeedsWith(testCode)
+          _ <- env.authPropertyGenerator.nextSessionId.succeedsWith(testSessionId)
+          _ <- env.securityService.mac.succeedsWith(testMac)
+          _ <- env.authPropertyGenerator.nextAccessToken.succeedsWith(testAccessToken)
+          _ <- env.authorizationCodeRepository.create.succeedsWith(())
+          _ <- env.sessionRepository.create.succeedsWith(())
+          _ <- env.conversationRepository.delete.succeedsWith(true)
+          result <- env.service.finishPasskeyEnroll(authId, recordWithUser, enrollStep, "resp", Some("My MacBook"))
```
[the remark this answers is not in the document]
Comment (BKJN1) [reply auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala]: added

File: auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala [hunk 2]
```diff
@@ -308,6 +308,71 @@ object PasskeyConversationServiceSpec extends UnitSpecBase:
           _ <- env.conversationRepository.delete.succeedsWith(true)
           result <- env.service.skipPasskey(authId, recordWithUser)
         yield assertTrue(result.isInstanceOf[ConversationResult.Complete])
-      }
-    )
+      },
+      test("return IllegalState for empty name") {
```
[the remark this answers is not in the document]
Comment (BKJN1) [reply auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala]: fixed

File: auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala [hunk 3]
```diff
@@ -308,6 +308,71 @@ object PasskeyConversationServiceSpec extends UnitSpecBase:
...
+        val env = Env()
+        val recordWithUser = baseRecord.copy(userId = Some(userId))
+        val enrollStep = ConversationStep.PasskeyEnroll("reg-req", "{}")
+        for
+          result <- env.service.finishPasskeyEnroll(authId, recordWithUser, enrollStep, "resp", Some(""))
```
[the remark this answers is not in the document]
Comment (BKJN1) [reply auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala]: added as a new test

File: auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala [hunk 4]
```diff
@@ -309,5 +382,5 @@ object PasskeyConversationServiceSpec extends UnitSpecBase:
           result <- env.service.skipPasskey(authId, recordWithUser)
         yield assertTrue(result.isInstanceOf[ConversationResult.Complete])
       }
-    )
   )
+)
```
[the remark this answers is not in the document]
Comment (BKJN1) [reply auth/src/test/scala/versola/oauth/conversation/PasskeyConversationServiceSpec.scala]: fixed

**Comment by BKJN1** [comment]
@Augment review

File: auth/src/main/scala/versola/oauth/conversation/ConversationService.scala [hunk 5]
```diff
@@ -491,18 +491,24 @@ object ConversationService:
         response: String,
         name: Option[String],
     ): Task[ConversationResult.Render] =
-      conversation.userId match
-        case None =>
+      val validatedName: Option[String] = name.map(_.trim)
+
+      validatedName match
+        case Some(n) if n.isEmpty || n.length > 64 || !n.matches("[\\p{L}\\p{N} _\\-'.]+") =>
```
[the remark this answers is not in the document]
Comment (BKJN1) [reply auth/src/main/scala/versola/oauth/conversation/ConversationService.scala]: fixed

**Comment by BKJN1** [comment]
augment review

**Comment by goshacodes** [comment]
Closing this in favor of #79
